File: app/src/main/java/org/odddev/fantlab/core/rx/ISchedulersResolver.java
package org.odddev.fantlab.core.rx;

import rx.Scheduler;
import rx.Single;

/**
 * @author kenrube
 * @since 23.08.16
 */

public interface ISchedulersResolver {

    Scheduler ioScheduler();

    Scheduler mainThreadScheduler();

    <T> Single.Transformer<T, T> applyDefaultSchedulers();
}

File: app/src/main/java/org/odddev/fantlab/core/rx/ISchedulersResolver.kt
package org.odddev.fantlab.core.rx

import rx.Scheduler
import rx.Single

/**
 * @author kenrube
 * *
 * @since 23.08.16
 */

interface ISchedulersResolver {

	fun ioScheduler(): Scheduler

	fun mainThreadScheduler(): Scheduler

	fun <T> applyDefaultSchedulers(): Single.Transformer<T, T>
}
